packages/flutter/lib/src/widgets/will_pop_scope.dart

```diff
@@ -9,10 +9,87 @@ import 'routes.dart';
 /// Registers a callback to veto attempts by the user to dismiss the enclosing
 /// [ModalRoute].
 ///
+/// {@tool snippet --template=stateful_widget}
+///
+/// Whenever the back button is pressed, you will get a callback at [onWillPop],
+/// which returns a [Future]. If the [Future] returns true, the screen is
+/// popped.
+///
+/// ```dart
+/// bool shouldPop = true;
+/// @override
+/// Widget build(BuildContext) {
+///   return WillPopScope (
+///     onWillPop: () async {
+///       return shouldPop;
+///     },
+///     child: Center(
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_material}
+/// ```dart
+/// bool shouldPop = true;
+/// @override
+/// Widget build(BuildContext context) {
+///   return WillPopScope(
+///     onWillPop: () async {
+///       return shouldPop;
+///     },
+///     child: Scaffold(
+///       appBar: AppBar(
+///         title: Text("Flutter WillPopScope demo"),
+///       ),
+///       body: Center(
+///         child: Wrap(
+///           spacing: 20,
+///           direction: Axis.vertical,
+///           crossAxisAlignment: WrapCrossAlignment.center,
+///           children: [
+///             OutlinedButton(
+///               child: Text('Push'),
+///               onPressed: () {
+///                 Navigator.of(context).push(
+///                   MaterialPageRoute(
+///                     builder: (context) {
+///                       return MyStatefulWidget();
+///                     },
+///                   ),
+///                 );
+///               },
+///             ),
+///             OutlinedButton(
+///               child: Text('Pop'),
+///               onPressed: () {
+///                 setState(
+///                   () {
+///                     shouldPop = !shouldPop;
+///                   },
+///                 );
+///               },
+///             ),
+///             Text("Current value of shouldPop is $shouldPop."
+///                 "Push to a new screen and then tap on the back button in appbar to toggle shouldPop"),
```

Comment:
"Current value" -> "The current value"

Reply:
This is still over 80 columns, so it will look odd in DartPad. You could add another break to this line:

```
///             Text("The current value of shouldPop is $shouldPop."
///                 "Push to a new screen and then tap on the "
///                 "back button in appbar to toggle shouldPop"),
```